src/ray/object_manager/ownership_based_object_directory.h

```diff
@@ -126,7 +132,11 @@ class OwnershipBasedObjectDirectory : public IObjectDirectory {
   std::function<void(const ObjectID &, const rpc::ErrorType &)> mark_as_failed_;
 
   /// A buffer for batch object location updates.
-  absl::flat_hash_map<WorkerID, absl::flat_hash_map<ObjectID, rpc::ObjectLocationState>>
+  /// Since absl::flat_hash_map doesn't maintain the insertion order,
+  /// we use a deque here to achieve FIFO.
```

Comment:
Can you comment about the mapping a little bit? Maybe 

owner id -> {(FIFO object queue (to avoid starvation), map for the latest state of objects)}